homeassistant/components/zha/core/gateway.py

```diff
@@ -766,7 +767,18 @@ async def shutdown(self) -> None:
             unsubscribe()
         for device in self.devices.values():
             device.async_cleanup_handles()
-        await self.application_controller.shutdown()
+        if (
+            hasattr(self, "application_controller")
```

Comment:
I'd add a comment why the hasattr check is needed.

Reply:
done